api/client/build.go

```diff
@@ -170,10 +168,11 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	}
 
 	var resolvedTags []*resolvedTag
-	if isTrusted() {
+	pullBehavior, translator := cli.trustedPullBehavior(flPull.Val())
+	if translator != nil {
```

Comment:
I'd move this check inside `replaceDockerfileTarWrapper` rather than relying on callers to do it for you. It would just return the context argument unmodified in the case translator was nil. Not an issue for now but might prevent somebody else a headache in the future.